File: src/main/java/course/examples/delegates/java/PersonDao.java
package course.examples.delegates.java;

import course.examples.delegates.Person;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface PersonDao {
    @Nullable
    public Person getPerson(int id);
    public List<Person> getPersons();
    public void insertPerson(Person person);
    public void updatePerson(Person person);
    public void deletePerson(Person person);
}

File: src/main/java/course/examples/delegates/java/PersonDao.kt
package course.examples.delegates.java

import course.examples.delegates.Person
import org.jetbrains.annotations.Nullable

interface PersonDao {
    @Nullable
    fun getPerson(id: Int): Person
    fun getPersons(): List<Person>
    fun insertPerson(person: Person)
    fun updatePerson(person: Person)
    fun deletePerson(person: Person)
}
